### PR TITLE
[hadoop][datanode] Add datanode data stream for hadoop

### PR DESCRIPTION
**Pull request by yug-elastic**
## What does this PR do?

- Generated the skeleton of Hadoop integration package.
- Added 1 data stream ( DataNode Metrics )
- Added data collection logic.
- Added the ingest pipelines.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added system test cases.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^8.0.0`).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/hadoop directory.
- Run the following command to run tests. 
> elastic-package test

### CODE REVIEW

File: packages/hadoop/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+title: "Hadoop"
+version: 0.1.0
+license: basic
+description: "This Elastic integration collects metrics from hadoop."
+type: integration
+categories:
+  - containers
+  - datastore
+conditions:
+  kibana.version: "^8.0.0"
```
Comment (yug-elastic) [inline packages/hadoop/manifest.yml:12]: This will be updated with 8.2.0 after testing this integration on 8.2.0.

**Comment by yug-elastic** [comment]
This PR is a split of https://github.com/elastic/integrations/pull/2614 as discussed over the comment https://github.com/elastic/integrations/pull/2614#issuecomment-1082704609

File: packages/hadoop/data_stream/datanode/fields/fields.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+- name: hadoop.datanode
+  type: group
+  release: beta
```
Comment (mtojek) [inline packages/hadoop/data_stream/datanode/fields/fields.yml:3]: Release tag to be removed.

File: packages/hadoop/data_stream/datanode/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    description: Collect DataNode metrics
+    template_path: stream.yml.hbs
+    vars:
+      - name: hosts
+        type: text
+        title: Hosts
+        multi: true
+        required: true
+        show_user: true
+        description: List of Hosts.
```
Comment (mtojek) [inline packages/hadoop/data_stream/datanode/manifest.yml:15]: nit: hosts (lower case)

File: packages/hadoop/data_stream/datanode/sample_event.json [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                "capacity": 48420556800,
+                "remaining": 11991523328
+            },
+            "estimated_capacity_lost_total": 0,
+            "last_volume_failure_date": "2022-01-01T00:00:00.000Z",
+            "num_blocks_cached": 0,
+            "num_blocks_failed": {
+                "to_cache": 0,
+                "to_uncache": 0
+            },
```
Comment (mtojek) [inline packages/hadoop/data_stream/datanode/sample_event.json:50]: I would suggest considering a bit different structure:

```
blocks {
   cached: {
       total: N
   }
   failed: {
        total: X
        to_cache: Y,
        to_cache: Z
   }
}
```

**Review by mtojek — APPROVED** [review]
LGTM!

File: packages/hadoop/manifest.yml (hunk 1 above)
Comment (yug-elastic) [reply packages/hadoop/manifest.yml]: Kibana version has been updated to 8.1.0 as discussed over weekly sync for all the integrations.

**Review by lalit-satapathy — APPROVED** [review]

**Pull request merged**